core/src/main/scala/scalaz/std/String.scala

```diff
@@ -48,9 +48,8 @@ trait StringFunctions {
 
   // Parsing functions.
   private def intChars(s: String): Boolean = (s.length > 0) && (s.forall(c => c.isDigit || c == '-'))
-  private def floatChars(s: String): Boolean = (s.length > 0) && (s.forall(c => c.isDigit || c == '-' || c == '.'))
```

Comment:
`floatChars`, in the same maner as `intChars`, need to be fixed, but not removed.